embeddings/mlx_embeddings.py

```diff
@@ -0,0 +1,205 @@
+        self.ln1 = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+        self.ln2 = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+        self.linear1 = nn.Linear(config.hidden_size, config.intermediate_size)
+        self.linear2 = nn.Linear(config.intermediate_size, config.hidden_size)
+        self.gelu = nn.GELU()
+
+    def __call__(self, x, mask):
+        attention_out = self.attention(x, x, x, mask)
+        add_and_norm = self.ln1(x + attention_out)
+
+        ff = self.linear1(add_and_norm)
+        ff_gelu = self.gelu(ff)
+        ff_out = self.linear2(ff_gelu)
+        x = self.ln2(ff_out + add_and_norm)
+
+        return x
+
+class TransformerEncoder(nn.Module):
+    def __init__(self, config: BertConfig):
+        super().__init__()
+        self.layers = [
+            TransformerEncoderLayer(config)
+            for i in range(config.num_hidden_layers)
+        ]
+
+    def __call__(self, x, mask):
+        for layer in self.layers:
+            x = layer(x, mask)
+        return x
+
+class BertEmbeddings(nn.Module):
+    def __init__(self, config: BertConfig):
+        self.word_embeddings = nn.Embedding(config.vocab_size, config.hidden_size)
+        self.token_type_embeddings = nn.Embedding(2, config.hidden_size)
+        self.position_embeddings = nn.Embedding(
+            config.max_position_embeddings, config.hidden_size
+        )
+        self.norm = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+
+    def __call__(self, input_ids: mx.array, token_type_ids: mx.array) -> mx.array:
+        words = self.word_embeddings(input_ids)
+        position = self.position_embeddings(
+            mx.broadcast_to(mx.arange(input_ids.shape[1]), input_ids.shape)
+        )
+        token_types = self.token_type_embeddings(token_type_ids)
+
+        embeddings = position + words + token_types
+        return self.norm(embeddings)
+
+
+class Bert(nn.Module):
+    def __init__(self, config: BertConfig):
+        self.embeddings = BertEmbeddings(config)
+        self.encoder = TransformerEncoder(config)
+        self.pooler = nn.Linear(config.hidden_size, config.hidden_size)
+
+    def __call__(
+        self,
+        input_ids: mx.array,
+        token_type_ids: mx.array = None,
+        attention_mask: mx.array = None,
+    ) -> tuple[mx.array, mx.array]:
+        if token_type_ids is None:
+            # just make it all 0s
+            token_type_ids = mx.zeros_like(input_ids)
+        x = self.embeddings(input_ids, token_type_ids)
+
+        if attention_mask is not None:
+            # convert 0's to -infs, 1's to 0's, and make it broadcastable
+            attention_mask = mx.log(attention_mask)
+            attention_mask = mx.expand_dims(attention_mask, (1, 2))
+
+        y = self.encoder(x, attention_mask) # shape: B, L, D
+        return y, mx.tanh(self.pooler(y[:, 0])) # shape: B, D
+    
+    @classmethod
+    def from_hugging_face(cls, model_path: str, precision_nbits: int = 8):
```

Comment:
Quick comment: Can this just be `from_pretrained`? It sounds more in sync with HF lingo and would also feel more familiar to people.